### PR TITLE
[Enhancement] Imporve or predicate statistics check

### PR DESCRIPTION
**Pull request by Seaven**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

core modify:
1. update PredicateStatisticsCalculator
   a. update DisConsecutiveOr check
   b. Fix or statistics compute bug(line: 292)

other:
1. update NormalizePredicateRule.inPredicate rule, only extract variable column
2. update planner profile, add level and order, support repeat use
3. rename PredicateStatisticsCalculatingVisitor 
4. remove unuse code

## Problem Summary(Required) ：

2000 predicate query cost 40s -> 8s, other bottlenecks:
1. Expr hashCode and equals, repeat a lot of calculations (1.8s)
2. NormalizePredicateRule.compound, repeat a lot of calculations(3s)
3. StackOverflow

it's hard to modify

```
    1691ms|-- Parser[1] 436ms
    2129ms|-- Planner[1] 7652ms
    2134ms|    -- Analyzer[1] 35ms
    2179ms|    -- Transformer[1] 4081ms
    2225ms|        -- filter1[1] 4023ms
    2225ms|            -- translate[1] 3589ms
    2238ms|                -- HashEquals[16005] 1749ms
    4010ms|                -- rewritePredicate[2] 1805ms
    5815ms|            -- rewriteSubquery[1] 433ms
    6261ms|    -- Optimizer[1] 3453ms
    6358ms|        -- Optimizer.RuleBaseOptimize[1] 3202ms
    6363ms|            -- PushDownPredicate[1] 3049ms
    6928ms|                -- RangeRewriter[1] 693ms
    7621ms|                -- rewritePredicate[1] 1768ms
    9566ms|        -- Optimizer.CostBaseOptimize[1] 127ms
    9582ms|            -- predicate statistics[2] 90ms
    9695ms|        -- Optimizer.PhysicalRewrite[1] 19ms
    9715ms|    -- ExecPlanBuild[1] 67ms
    9782ms|-- Test View[1] 6113ms
```

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

**Review by kangkaisen — DISMISSED** [review]

**Review by kangkaisen — DISMISSED** [review]

**Comment by Seaven** [comment]
run https://github.com/StarRocks/starrocks/pull/15697

**Comment by Seaven** [comment]
run starrocks_fe_unittest

**Comment by Seaven** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 47 / 53 (88.68%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/PlannerProfile.java|23|29|79.31%|[79, 88, 241, 242, 249, 250]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rewrite/scalar/NormalizePredicateRule.java|10|10|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/PredicateStatisticsCalculator.java|12|12|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/StatementPlanner.java|2|2|100.00%|[]|

**Review by kangkaisen — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/scalar/NormalizePredicateRule.java [hunk 1]
```diff
@@ -164,15 +164,27 @@ public ScalarOperator visitInPredicate(InPredicateOperator predicate, ScalarOper
         ScalarOperator lhs = predicate.getChild(0);
         boolean isIn = !predicate.isNotIn();
 
-        for (ScalarOperator child : predicate.getChildren().subList(1, predicate.getChildren().size())) {
+        List<ScalarOperator> constants = predicate.getChildren().stream().skip(1).filter(ScalarOperator::isConstant)
```
Comment (HangyuanLiu) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/scalar/NormalizePredicateRule.java:167]: expr "col in (var+1, 2)" may be missing

**Review by HangyuanLiu — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by Seaven** [comment]
run starrocks_admit_test

**Comment by Seaven** [comment]
run starrocks_admit_test

**Comment by Seaven** [comment]
@mergify backport branch-2.5

**Pull request merged**